Framework/src/Checker.cxx

```diff
@@ -37,22 +39,70 @@ using namespace o2::configuration;
 using namespace o2::monitoring;
 using namespace o2::quality_control::core;
 using namespace o2::quality_control::repository;
+using namespace o2::quality_control::monitor;
 
 namespace o2::quality_control::checker
 {
 
+/// Static functions
+o2::header::DataDescription Checker::createCheckerDataDescription(const std::string taskName)
+{
+  if (taskName.empty()) {
+    BOOST_THROW_EXCEPTION(FatalException() << errinfo_details("Empty taskName for checker's data description"));
+  }
+  o2::header::DataDescription description;
+  description.runtimeInit(std::string(taskName.substr(0, o2::header::DataDescription::size - 4) + "-chk").c_str());
+  return description;
+}
+
+o2::framework::Inputs Checker::createInputSpec(const std::string checkName, const std::string configSource)
+{
+  std::unique_ptr<ConfigurationInterface> config = ConfigurationFactory::getConfiguration(configSource);
+  o2::framework::Inputs inputs;
+  for(auto& [key, sourceConf]: config->getRecursive("qc.check."+checkName+".dataSource")){
+    (void)key;
+    if(sourceConf.get<std::string>("type") == "Task"){
+      const std::string& taskName = sourceConf.get<std::string>("name"); 
+      QcInfoLogger::GetInstance() << ">>>> Check name : " << checkName << " input task name: " << taskName << " " << TaskRunner::createTaskDataDescription(taskName).as<std::string>() << AliceO2::InfoLogger::InfoLogger::endm;
+      o2::framework::InputSpec input{taskName, TaskRunner::createTaskDataOrigin(),  TaskRunner::createTaskDataDescription(taskName)};
+      inputs.push_back(std::move(input));
+    }
+  }
+
+  return inputs;
+}
+
+std::string Checker::createCheckerIdString(){
+  return std::string("QC-CHECKER");
+}
+
+
+/// Members 
+
 // TODO do we need a CheckFactory ? here it is embedded in the Checker
 // TODO maybe we could use the CheckerFactory
 
-Checker::Checker(std::string checkerName, std::string taskName, std::string configurationSource)
-  : mCheckerName(checkerName),
+
+Checker::Checker(std::vector<std::string> checkerNames, std::string configurationSource)
+  : mDeviceName(createCheckerIdString() + "-" + checkerNames.front()),
+    mCheckerNames{checkerNames},
```

Comment:
Yes. Should I change the naming to mCheckNames?
Also, what should be the name of the device? I am not sure if one of the check names is correct. Might mislead in the structure.

Reply:
> Yes. Should I change the naming to mCheckNames?

Yes please. There are more names in the code which have "checker" instead of "check", for example in the QualityObject, runnerUtilts, infrastructureGenerator. Could you also take care of them?
> Also, what should be the name of the device? I am not sure if one of the check names is correct. Might mislead in the structure.

Good question. The only thing they have in common is inputs, though listing them in the name isn't the best idea. I would propose to give it a unique number at the end of a common name "Checker".